tasks/test-project/rebuild-test-project-fixture.js

```diff
@@ -45,479 +71,660 @@ const args = yargs(hideBin(process.argv))
-  }
-
-  let skip = skipFn?.()
+// Check for existing indices from previously committed steps
+const existingIndices = []
+let resumeFromIndex = resumeStep
 
-  if (skip) {
-    if (typeof skip === 'boolean' && skip) {
-      // if skip is just `true`, then we use the default skip message
-      skip = 'Skipping...'
-    }
-
-    tuiContent.update({
-      spinner: {
-        enabled: false,
-      },
-      header: `${RedwoodStyling.green('✔')} ${step}. ${title}`,
-      content: ' '.repeat(stepId.length + 4) + RedwoodStyling.info(skip) + '\n',
+if (resume) {
+  try {
+    const subprocess = execa.sync('git', ['log', '--oneline'], {
+      cwd: OUTPUT_PROJECT_PATH,
+      stdio: 'pipe',
     })
-
-    tui.stopReactive()
-
-    return
+    subprocess.stdout.split('\n').forEach((line) => {
+      const parts = line.split(' ')
+      existingIndices.push(parts[parts.length - 1])
+    })
+  } catch (error) {
+    console.error('Failed to get git log')
+    console.error(error)
+    process.exit(1)
   }
 
-  let promise
-
-  try {
-    promise = task()
-  } catch (e) {
-    // This code handles errors from synchronous tasks
-
-    tui.stopReactive(true)
-
-    if (e instanceof ExecaError) {
-      tui.displayError(
-        'Failed ' + title.toLowerCase().replace('...', ''),
-        'stdout:\n' + e.stdout + '\n\n' + 'stderr:\n' + e.stderr
-      )
-    } else {
-      tui.displayError(
-        'Failed ' + title.toLowerCase().replace('...', ''),
-        e.message
-      )
-    }
-
-    process.exit(e.exitCode)
+  // Start from the last index if no step is given
+  if (resumeStep === undefined) {
+    // git log will be sorted by time so that index 0 should be the latest commit and hence the last step
+    resumeFromIndex = existingIndices[0].replace('x', '')
   }
+}
 
-  if (isAwaitable(promise)) {
-    const result = await promise.catch((e) => {
-      // This code handles errors from asynchronous tasks
-
-      tui.stopReactive(true)
-
-      if (e instanceof ExecaError) {
-        tui.displayError(
-          'Failed ' + title.toLowerCase().replace('...', ''),
-          'stdout:\n' + e.stdout + '\n\n' + 'stderr:\n' + e.stderr
-        )
-      } else {
-        tui.displayError(
-          'Failed ' + title.toLowerCase().replace('...', ''),
-          e.message
-        )
-      }
-
-      process.exit(e.exitCode)
-    })
-
-    if (Array.isArray(result)) {
-      const tuiTaskList = result
-      for (let i = 0; i < tuiTaskList.length; i++) {
-        // Recurse through all tasks
-        await tuiTask({
-          step: i,
-          ...tuiTaskList[i],
-          parent: stepId,
-        })
-      }
+function indexIsLower(index, comparisonIndex) {
+  const levels1 = index.split('.')
+  const levels2 = comparisonIndex.split('.')
+  // Loop through the levels until one index is exhausted
+  for (let i = 0; i < Math.max(levels1.length, levels2.length); i++) {
+    const level1 = parseInt(levels1[i]) || 0 // Use 0 if level is undefined or NaN
```

Comment:
Just to be safe :) 
```suggestion
    const level1 = parseInt(levels1[i], 10) || 0 // Use 0 if level is undefined or NaN
```
And nice! I didn't know `NaN` was falsey!